planner/core/exhaust_physical_plans.go

```diff
@@ -1460,10 +1479,74 @@ func (p *LogicalJoin) exhaustPhysicalPlans(prop *property.PhysicalProperty) ([]P
 	return joins, true
 }
 
+func (p *LogicalJoin) tryToGetBroadCastJoin(prop *property.PhysicalProperty) []PhysicalPlan {
+	/// todo remove this restriction after join on new collation is supported in TiFlash
+	if collate.NewCollationEnabled() {
+		return nil
+	}
+	if !prop.IsEmpty() {
+		return nil
+	}
+	if prop.TaskTp != property.RootTaskType && !prop.IsFlashOnlyProp() {
+		return nil
+	}
+
+	if p.JoinType != InnerJoin || len(p.LeftConditions) != 0 || len(p.RightConditions) != 0 || len(p.OtherConditions) != 0 || len(p.EqualConditions) == 0 {
```

Comment:
It's because tiflash don't support join with filter, but for inner join, left cond and right cond are pushed down, other condition is a problem and should be treated as a filter in tiflash, this will refine in next pr.